src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocParagraphCheck.java

```diff
@@ -42,8 +42,14 @@
  * <li>Each paragraph but the first has &lt;p&gt; immediately
  * before the first word, with no space after.</li>
  * <li>First paragraph tag should not precede
- * <a href="https://www.w3schools.com/html/html_blocks.asp">HTML block-tag</a>,
- * nested paragraph tags are allowed to do that.</li>
+ * <a href="https://www.w3schools.com/html/html_blocks.asp">HTML block-tag</a>.
+ * Javadoc support only following block-tags:
```

Comment:
The doc says:

```
First paragraph tag should not precede HTML block-tag
```

Here `First paragraph tag` can confuse people. By first paragraph tag, we meant the outer most paragraph tag, not the paragraph tag which comes first in the javadoc. User might misunderstand this statement as:

```
The paragraph tag which occurs first in the javadoc should not be preceded by block-tag, it's sibling paragraph tag can do that
```

The check's behavior:

```java
/**
 * Plugin to standardize all directions to their appropriate abbreviation.
 *
 * <p><div>Test</div></p>
 *
 * <p><h1>Test</h1></p>
 *
 * <table>
 * <tbody>
 * <p>
 * <tr>
 * nested paragraph preceding block tag, this is ok.
 * </tr>
 * </p>
 * </tbody>
 * </table>
 */
public class Test {
  /** Some javadoc. */
  public static void main(String[] args) {

  }
}
```

```
$  java -jar .\checkstyle-10.20.1-all.jar -c .\javadocparagraph_config.xml .\Test.java
Starting audit...
[ERROR] C:\checkstyle testing\.\Test.java:4:4: <p> tag should not precede HTML block-tag '<div>', <p> tag should be removed. [JavadocParagraph]
[ERROR] C:\checkstyle testing\.\Test.java:6:4: <p> tag should not precede HTML block-tag '<h1>', <p> tag should be removed. [JavadocParagraph]
Audit done.
Checkstyle ends with 2 errors.
```

Here the user might not expect violation on:

```
<p><h1>Test</h1></p>
```

We should update the doc to make it more clear, something like:

```
The outer most paragraph tags should not preceded HTML block-level tags. Nested paragraph tags are allowed to do that.
```

Reply:
done.